modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition_manager.py

```diff
@@ -51,6 +57,67 @@ def wait_partitions(cls, partitions):
             [block for partition in partitions for block in partition.list_of_blocks]
         )
 
+    @classmethod
+    @_inherit_docstrings(
+        GenericRayDataframePartitionManager.split_pandas_df_into_partitions
+    )
+    def split_pandas_df_into_partitions(
+        cls, df, row_chunksize, col_chunksize, update_bar
+    ):
+        # it was found out, that with the following condition it's more beneficial
+        # to use the distributed splitting, let's break them down:
+        #   1. The distributed splitting is used only when there's more than 6mln elements
+        #   in the `df`, as with fewer data it's better to use the sequential splitting
+        #   2. Only used with numerical data, as with other dtypes, putting the whole big
+        #   dataframe into the storage takes too much time.
+        #   3. The distributed splitting consumes more memory that the sequential one.
+        #   It was estimated that it requires ~2.5x of the dataframe size, for now there
+        #   was no good way found to automatically fall back to the sequential
+        #   implementation in case of not enough memory, so currently we're enabling
+        #   the distributed version only if 'AsyncReadMode' is set to True. Follow this
+        #   discussion for more info on why automatical dispatching is hard:
+        #   https://github.com/modin-project/modin/pull/6640#issuecomment-1759932664
+        enough_elements = (len(df) * len(df.columns)) > 6_000_000
+        all_numeric_types = all(is_numeric_dtype(dtype) for dtype in df.dtypes)
+        async_mode_on = AsyncReadMode.get()
```

Comment:
`from_pandas()` is technically related to IO and so it can also be viewed as reading. In my opinion, the introduced logic fits to the term "async reading". Moreover, it seems that we already have too much configs, so grouping related features under one name seems reasonable

Reply:
> grouping related features under one name seems reasonable

If a config suites the concrete features, I agree.

> In my opinion, the introduced logic fits to the term "async reading"

Could you elaborate how the introduced logic fits to the term "async reading"?

Reply:
Well, in modin docs async reading defined as:
https://github.com/modin-project/modin/blob/b0ef411df7dec628d4d9fc035f60c9e74895518f/modin/config/envvars.py#L661

I would expand the definition to that `pd.DataFrame` constructor only launches tasks for the dataframe to be created, but not ensures that the construction is finalized by the time the reading function returns a dataframe.

And this is exactly what's happening with this distributed implementation. With the new implementation, `pd.DataFrame(pandas_df)` call only launches a bunch of tasks and doesn't wait for them to be completed, which results in that the result of `pd.DataFrame(...)` is still computing something.

The async reading can cause certain problems, like the one [described in this comment](https://github.com/modin-project/modin/pull/6640#issuecomment-1759932664), when you read many dataframes in a for-loop it can lead to excessive memory consumption since multiple dataframes are being created at the same time.

Reply:
Okay, makes sense to keep this config only but let's expand the docstring of the config with respect to possible sources and problems that may occur.

Reply:
done